tests/test_collector.py

```diff
@@ -59,15 +59,45 @@ def test_extract_metrics():
 @responses.activate
 def test_get_azure_data():
 
-    enrollment='12345'
+    enrollment = '123'
+    base_url = "https://ea.azure.com/rest/{}/usage-report".format(enrollment)
+    params = "?month={}&type=detail&fmt=Json".format(current_month())
+
     c = AzureEABillingCollector('cloud_costs', enrollment, 'abc123xyz')
 
     responses.add(
         method='GET',
-        url="https://ea.azure.com/rest/{}/usage-report?month=2017-03&type=detail&fmt=Json".format(enrollment),
+        url=base_url+params,
         match_querystring=True,
         json=sample_data
     )
 
-    data = c._get_azure_data('2017-03')
+    data = c._get_azure_data(current_month())
     assert data == sample_data
+
+
+@responses.activate
+def test_empty_month():
+    """
+    If no usage details have are available for a given month the API does not return a JSON document.    
+    """
+    enrollment = '123'
+    base_url = "https://ea.azure.com/rest/{}/usage-report".format(enrollment)
+    params = "?month={}&type=detail&fmt=Json".format(current_month())
+
+    c = AzureEABillingCollector('cloud_costs', enrollment, 'abc123xyz')
+
+    empty_month_data = """"Usage Data Extract",
```

Comment:
Same comment as above. Two things to think about:
* If you think that both the application and the collector tests benefit from the full legit response, extract it to a shared constant (in test_helpers.py) called `AZURE_OUTPUT_FOR_EMPTY_MONTH` or something.
* Probably you only need the full output (if you need it) in the collector test. The application test could go with `"not a dictionary"` as test input. That would require changing the code to check for an exception.